### PR TITLE
Cleaned up foreign key cleanup for user group deletion

### PR DESCRIPTION
**Pull request by hagen-danswer**

### CODE REVIEW

File: backend/ee/danswer/db/user_group.py [hunk 1]
```diff
@@ -33,6 +34,93 @@
 logger = setup_logger()
```
Comment (hagen-danswer) [inline backend/ee/danswer/db/user_group.py:36]: just moved these to the top of the file

File: backend/ee/danswer/db/user_group.py [hunk 2]
```diff
@@ -476,21 +528,6 @@ def update_user_group(
...
-            TokenRateLimit__UserGroup.user_group_id == user_group_id
-        )
-    ).all()
-    for (
-        token_rate_limit__user_group_relationship
-    ) in token_rate_limit__user_group_relationships:
-        db_session.delete(token_rate_limit__user_group_relationship)
-
-
 def prepare_user_group_for_deletion(db_session: Session, user_group_id: int) -> None:
```
Comment (hagen-danswer) [inline backend/ee/danswer/db/user_group.py:531]: This function now cleans up all 7 fk relationships for usergroups:

- credential
- user
- llm_provider
- document_set
- token_rate_limit
- persona
- connector_credential_pair

File: backend/ee/danswer/db/user_group.py [hunk 3]
```diff
@@ -517,31 +569,12 @@ def prepare_user_group_for_deletion(db_session: Session, user_group_id: int) ->
...
-def _cleanup_document_set__user_group_relationships__no_commit(
-    db_session: Session, user_group_id: int
-) -> None:
-    """NOTE: does not commit the transaction."""
-    db_session.execute(
-        delete(DocumentSet__UserGroup).where(
-            DocumentSet__UserGroup.user_group_id == user_group_id
-        )
-    )
+def delete_user_group(db_session: Session, user_group: UserGroup) -> None:
```
Comment (hagen-danswer) [inline backend/ee/danswer/db/user_group.py:572]: removed duplicate fk cleanup  so this function is now super simple (but it assumes fk cleanup has already been performed

File: backend/ee/danswer/db/user_group.py [hunk 4]
```diff
@@ -553,29 +586,6 @@ def mark_user_group_as_synced(db_session: Session, user_group: UserGroup) -> Non
     db_session.commit()
 
 
-def delete_user_group(db_session: Session, user_group: UserGroup) -> None:
```
Comment (hagen-danswer) [inline backend/ee/danswer/db/user_group.py:556]: this function was moved up to be near the fk_cleanup function

File: backend/ee/danswer/db/user_group.py [hunk 5]
```diff
@@ -517,31 +569,12 @@ def prepare_user_group_for_deletion(db_session: Session, user_group_id: int) ->
...
-    db_session.execute(
-        delete(DocumentSet__UserGroup).where(
-            DocumentSet__UserGroup.user_group_id == user_group_id
-        )
-    )
+def delete_user_group(db_session: Session, user_group: UserGroup) -> None:
+    """
+    This assumes that all the fk cleanup has already been done.
+    """
+    db_session.delete(user_group)
```
Comment (rkuo-danswer) [inline backend/ee/danswer/db/user_group.py:576]: The old function had an explicit comment about flushing, but that doesn't seem to be getting handled here.  Any reason why?

**Review by rkuo-danswer — CHANGES_REQUESTED** [review]
Possible to add tests for these relationships?  As an example I just injected an entry in the DB for connector deletion ... maybe we could do something similar here.

**Review by rkuo-danswer — APPROVED** [review]

**Pull request merged**